hook.php

```diff
@@ -121,15 +121,25 @@ function plugin_credit_get_datas(NotificationTargetTicket $target)
     $ticket->getFromDB($id);
     $entity_id = $ticket->fields['entities_id'];
 
-    $query = <<<SQL
-        SELECT
-            `glpi_plugin_credit_entities`.`name`,
-            `glpi_plugin_credit_entities`.`quantity`,
-            (SELECT SUM(`glpi_plugin_credit_tickets`.`consumed`) FROM `glpi_plugin_credit_tickets` WHERE `glpi_plugin_credit_tickets`.`plugin_credit_entities_id` = `glpi_plugin_credit_entities`.`id` AND `glpi_plugin_credit_tickets`.`tickets_id` = {$id}) AS `consumed_on_ticket`,
-            (SELECT SUM(`glpi_plugin_credit_tickets`.`consumed`) FROM `glpi_plugin_credit_tickets` WHERE `glpi_plugin_credit_tickets`.`plugin_credit_entities_id` = `glpi_plugin_credit_entities`.`id`) AS  `consumed_total`
-        FROM `glpi_plugin_credit_entities`
-        WHERE `is_active`=1 and `entities_id`={$entity_id}
-SQL;
+    $query = [
+        'SELECT' => [
+            'name',
+            'quantity',
+            'consumed_on_ticket' => new QueryExpression(
```

Comment:
Either use a subquery, or use the iterator to build the query you give as expression